<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
    <arg name="depth" default="true"/>
    <arg name="from_kinect" default="false"/>
    <arg name="bkg_sep_mode" default="depth_auto" />
    <arg name="video_dir" value="/home/alex/MAS/CV/project/cv_nxtdrive/data/1"/>

    <group if="$(arg depth)">
        <group if="$(arg from_kinect)">
            <include file="$(find openni_launch)/launch/openni.launch"/>
            <node name="set_dr" pkg="dynamic_reconfigure" type="dynparam" args="set /camera/driver depth_registration true" />
        </group>
        <group unless="$(arg from_kinect)">
            <include file="$(find simple_pcl)/launch/play_video_as_bag.launch">
                <arg name="rgb_video" value="$(arg video_dir)/hand_rgb.avi"/>
                <arg name="depth_video" value="$(arg video_dir)/hand_depth.avi"/>
            </include> 
        </group>
        <node name="depth_detection" type="track_hand.py" pkg="cv_nxtdrive" args="$(arg bkg_sep_mode)" output="screen" />
    </group>
    <group unless="$(arg depth)">
        <include file="$(find cv_nxtdrive)/launch/capture_video.launch" />
        <node name="skin_detection" type="skin_detection.py" pkg="cv_nxtdrive" output="screen" />
    </group>
</launch>
